### PR TITLE
Fixed NeonAdapter::dump() for data with simple \Nette\DI\Statement

### PR DESCRIPTION
**Pull request by sallyx**
There is still a bug with following neon statements:
- ent(2)inner(3, 4)
- ent(3)::inner(5)
 
I have added a test which FAILS because of those statements.

Documentation - not needed
BC break - no

### CODE REVIEW

File: src/DI/Config/Adapters/NeonAdapter.php [hunk 1]
```diff
@@ -91,6 +91,13 @@ public function dump(array $data)
 			}
 			$tmp[$name] = $secData;
 		}
+		\array_walk_recursive($tmp, function(&$val) {
```
Comment (dg) [inline src/DI/Config/Adapters/NeonAdapter.php]: `array_walk_recursive($tmp, function (& $val)`

File: src/DI/Config/Adapters/NeonAdapter.php [hunk 2]
```diff
@@ -91,6 +91,13 @@ public function dump(array $data)
...
+			if($val instanceof \Nette\DI\Statement) {
```
Comment (dg) [inline src/DI/Config/Adapters/NeonAdapter.php]: `if ($val instanceof Statement)`

File: src/DI/Config/Adapters/NeonAdapter.php [hunk 3]
```diff
@@ -91,6 +91,13 @@ public function dump(array $data)
...
+				/* @var $val Neon\Entity */
```
Comment (dg) [inline src/DI/Config/Adapters/NeonAdapter.php]: remove comment

File: tests/php-unix.ini [hunk 4]
```diff
@@ -1,5 +1,6 @@
 [PHP]
 ;extension_dir = "./ext"
+extension=tokenizer.so
```
Comment (dg) [inline tests/php-unix.ini]: why?

Comment (sallyx) [inline tests/php-unix.ini]: Without this I get a lot of errors: Fatal error: Call to undefined function Nette\DI\token_get_all() in ...nette/di/src/DI/PhpReflection.php on line 117

Comment (dg) [inline tests/php-unix.ini]: Ok, but put it to a different commit.

**Comment by sallyx** [comment]
Tests will pass with nette/neon#21

File: src/DI/Config/Adapters/NeonAdapter.php [hunk 5]
```diff
@@ -91,6 +91,25 @@ public function dump(array $data)
 			}
 			$tmp[$name] = $secData;
 		}
+		\array_walk_recursive(
+			$tmp,
+			function (&$val) {
+				if($val instanceof Statement) {
```
Comment (dg) [inline src/DI/Config/Adapters/NeonAdapter.php]: `if (` with space

File: src/DI/Config/Adapters/NeonAdapter.php [hunk 6]
```diff
@@ -91,6 +91,25 @@ public function dump(array $data)
 			}
 			$tmp[$name] = $secData;
 		}
+		\array_walk_recursive(
```
Comment (dg) [inline src/DI/Config/Adapters/NeonAdapter.php]: please write global function without slash

File: src/DI/Config/Adapters/NeonAdapter.php [hunk 7]
```diff
@@ -91,6 +91,25 @@ public function dump(array $data)
...
+			$tmp,
+			function (&$val) {
```
Comment (dg) [inline src/DI/Config/Adapters/NeonAdapter.php]: `& $val` with space

**Comment by dg** [comment]
And now squash it to single commit please

**Comment by dg** [comment]
Try it with file `compiler.services.factory.neon`, it triggers `E_NOTICE: Trying to get property of non-object in Config\Adapters\NeonAdapter.php(102)`

Better syntax is IMHO `a()::b()` than `a()b()`

File: src/DI/Config/Adapters/NeonAdapter.php [hunk 8]
```diff
@@ -77,6 +77,33 @@ private function process(array $arr)
 		return $res;
 	}
 
+	/**
+	* @param Statement $val
```
Comment (dg) [inline src/DI/Config/Adapters/NeonAdapter.php]: please remove `@param` annotation

**Comment by dg** [comment]
Great! The last thing: change `"nette/neon": "~2.3"` to `"nette/neon": "^2.3.3"` in Composer and tests will pass (I hope).

File: tests/DI/NeonAdapter.phpt [hunk 9]
```diff
@@ -134,3 +134,76 @@ Assert::equal([
...
+		- @three
+		- foo
+
+	stdClass: stdClass
+	factory: Lorem
+	rich1: Lorem(1)foo()
+	rich2:
+		create: Lorem(Ipsum(@one))foo(1)
+
+	rich3: Factory::createLorem(1)foo()
```
Comment (dg) [inline tests/DI/NeonAdapter.phpt]: Is not better `Factory::createLorem(1)::foo()` than `Factory::createLorem(1)foo()`?

**Comment by dg** [comment]
Merged 3a378671b0a8c651179ffd68f9a587049686b1e1, thanks

File: tests/DI/NeonAdapter.phpt (hunk 9 above)
Comment (sallyx) [inline tests/DI/NeonAdapter.phpt]: Maybe, I didn't want to make the source code more complicated.
